tests/js/spec/views/organizationIntegrations/integrationListDirectory.spec.jsx

```diff
@@ -0,0 +1,260 @@
+      body,
+    });
+  });
+};
+
+describe('IntegrationListDirectory', function() {
+  beforeEach(function() {
+    Client.clearMockResponses();
+  });
+
+  const org = TestStubs.Organization();
+  const routerContext = TestStubs.routerContext();
+
+  describe('Renders view', function() {
+    it('shows installed integrations at the top', async function() {
+      mockResponse([
+        [
+          `/organizations/${org.slug}/config/integrations/`,
+          {
+            providers: [
+              {
+                canAdd: true,
+                canDisable: false,
+                features: ['commits', 'issue-basic'],
+                key: 'bitbucket',
+
+                metadata: {
+                  aspects: {},
+                  author: 'The Sentry Team',
+                  description:
+                    'Connect your Sentry organization to Bitbucket, enabling the following features:',
+
+                  features: [],
+                  issue_url:
+                    'https://github.com/getsentry/sentry/issues/new?title=Bitbucket%20Integration:%20&labels=Component%3A%20Integrations',
+                  noun: 'Installation',
+                  source_url:
+                    'https://github.com/getsentry/sentry/tree/master/src/sentry/integrations/bitbucket',
+                },
+                name: 'Bitbucket',
+
+                setupDialog: {
+                  height: 600,
+                  url: '/organizations/sentry/integrations/bitbucket/setup/',
+                  width: 600,
+                },
+                slug: 'bitbucket',
+              },
+            ],
+          },
+        ],
+        [
+          `/organizations/${org.slug}/integrations/`,
+          [
+            {
+              accountType: null,
+              configData: {},
+              configOrganization: [],
+              domainName: 'bitbucket.org/%7Bfb715533-bbd7-4666-aa57-01dc93dd9cc0%7D',
+              icon:
+                'https://secure.gravatar.com/avatar/8b4cb68e40b74c90427d8262256bd1c8?d=https%3A%2F%2Favatar-management--avatars.us-west-2.prod.public.atl-paas.net%2Finitials%2FNN-0.png',
+              id: '4',
+              name: '{fb715533-bbd7-4666-aa57-01dc93dd9cc0}',
+              provider: {
+                aspects: {},
+                canAdd: true,
+                canDisable: false,
+                features: ['commits', 'issue-basic'],
+                key: 'bitbucket',
+                name: 'Bitbucket',
+                slug: 'bitbucket',
+              },
+              status: 'active',
+            },
+          ],
+        ],
+        [
+          `/organizations/${org.slug}/sentry-apps/`,
+          [
+            {
+              allowedOrigins: [],
+              author: 'Sentry',
+              clientId:
+                'a6d35972d4164ef18845b1e2ca954fe70ac196e0b20d4d1e8760a38772cf6f1c',
+              clientSecret:
+                '8f47dcef40f7486f9bacfeca257022e092a483add7cf4d619993b9ace9775a79',
```

Comment:
@NisanthanNanthakumar is this your actual client secret lol? Let's not expose that in a test...

Reply:
@scefali it's a client secret for a fake internal app on my local instance

Reply:
Fair enough :)